system/Router/AutoRouter.php

```diff
@@ -126,18 +127,24 @@ public function getRoute(string $uri, string $httpVerb): array
             $controller .= $controllerName;
 
             $controller = strtolower($controller);
-
-            foreach ($this->protectedControllers as $controllerInRoute) {
-                if (! is_string($controllerInRoute)) {
-                    continue;
-                }
-                if (strtolower($controllerInRoute) !== $controller) {
-                    continue;
+            $methodName = strtolower($this->methodName());
+
+            foreach ($this->cliRoutes as $handler) {
+                if (is_string($handler)) {
+                    $handler = strtolower($handler);
+
+                    if (strpos($handler, $controller . '::' . $methodName) === 0) {
+                        throw new PageNotFoundException(
+                            'Cannot access CLI Route: ' . $uri
+                        );
+                    }
+
+                    if ($handler === $controller) {
+                        throw new PageNotFoundException(
+                            'Cannot access CLI Route: ' . $uri
+                        );
+                    }
                 }
-
-                throw new PageNotFoundException(
-                    'Cannot access the controller in a CLI Route. Controller: ' . $controllerInRoute
-                );
             }
```

Comment:
I restored the logic from v4.1.9:
https://github.com/codeigniter4/CodeIgniter4/blob/202f41ad522ba1d414b9d9c35aba1cb0c156b781/system/Router/Router.php#L492-L513